### PR TITLE
feat: add a logging interface

### LINKED ISSUE #152: Add logging interface
#### Reason/Context
We should provide a logging interface for users of the library to get notified on internal logs.

Use-case:
- In case two simplified models need to be merged some properties such as `$id` are overwritten, in these special cases I would like to inform the user that a clash between properties occurred and the output might not contain the expected output. 
- As a user I would like to see which models are generated

#### Description
I suggest we provide a logging interface so the users of the library can decide which logs they would like to do something with.

This could for example include 4 levels:
- Debug
- Info
- Warning
- Error

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR adds a logging interface to the library to allow logging for different scenarios.

**Related issue(s)**
fixes #152

### CODE REVIEW

File: src/utils/LoggingInterface.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  error(message?: any, ...optionalParams: any[]): void;
+}
+
+/**
+ * Logger class for the model generation library
+ * 
+ * This class acts as a forefront for any external loggers which is why it also implements the interface itself.
+ */
+export class Logger implements ModelLoggingInterface {
+  private static instance?: Logger;
```
Comment (magicmatatjahu) [inline src/utils/LoggingInterface.ts:18]: I don't think so this is a good idea. We should create default logger by in constructor, and avoid logic like `Logger.getInstance().*`. As developer I rather prefer `Logger.*`

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

File: src/utils/LoggingInterface.ts (hunk 1 above)
Comment (jonaslagoni) [reply src/utils/LoggingInterface.ts:18]: Hmm, good point, let me refactor it.

File: src/utils/LoggingInterface.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+   * Sets the logger to use for the model generation library
+   * 
+   * @param logger to add
+   */
+  setLogger(logger?: ModelLoggingInterface) {
+    this.logger = logger;
+  }
+}
+
+export const Logger: LoggerClass = new LoggerClass();
```
Comment (magicmatatjahu) [inline src/utils/LoggingInterface.ts:51]: Maybe by default we should set `Logger` as `this.logger`? 

```js
const Logger: LoggerClass = new LoggerClass();
Logger.setLogger(Logger);
export { Logger };
```

Comment (jonaslagoni) [reply src/utils/LoggingInterface.ts:51]: That would create an infinite loop 😄 The logger was designed for others to add their own logger implementation based on their own use case. We don't need to set the logger to itself. Maybe the naming of the class is a bit confusing 🤔?

Comment (magicmatatjahu) [reply src/utils/LoggingInterface.ts:51]: Ahh ok, I didn't check the infinity loop 😄 then LGTM.

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**